common-app-lib/src/cmr/common_app/services/kms_lookup.clj

```diff
@@ -240,32 +242,46 @@
+            _ (rl-util/log-redis-read-complete "lookup-by-short-name" kms-short-name-cache-key tm)
+            keywords (if keywords
+                       keywords
+                       (let [_ (load-cache-if-necessary context short-name-cache kms-short-name-cache-key)
+                             [tm kwords] (util/time-execution (hash-cache/get-value short-name-cache kms-short-name-cache-key keyword-scheme))]
+                         (rl-util/log-redis-read-complete "lookup-by-short-name" kms-short-name-cache-key tm)
+                         kwords))]
+        (get keywords (util/safe-lowercase short-name))))
+    (catch Exception e
+      (if (clojure.string/includes? (ex-message e) "Carmine connection error")
+        (do
+          (error "lookup-by-short-name found redis carmine exception. Will return nil result." e)
+          nil)
+        (throw e)))))
 
 (defn lookup-by-location-string
   "Takes a kms-index and a location string and returns the full KMS hierarchy for that location
   string. Returns nil if a keyword is not found. Comparison is made case insensitively."
   [context location-string]
-  (when-not (:ignore-kms-keywords context)
-    (let [location-cache (hash-cache/context->cache context kms-location-cache-key)
-          [tm keywords] (util/time-execution (hash-cache/get-value location-cache kms-location-cache-key (string/upper-case location-string)))
-          _ (rl-util/log-redis-read-complete "lookup-by-location-string" kms-location-cache-key tm)]
-      (if keywords
-        keywords
-        (let [_ (load-cache-if-necessary context location-cache kms-location-cache-key)
-              [tm kwords] (util/time-execution (hash-cache/get-value location-cache kms-location-cache-key (string/upper-case location-string)))]
-          (rl-util/log-redis-read-complete "lookup-by-location-string" kms-location-cache-key tm)
-          kwords)))))
+  (try
+    (when-not (:ignore-kms-keywords context)
+      (let [location-cache (hash-cache/context->cache context kms-location-cache-key)
+            [tm keywords] (util/time-execution (hash-cache/get-value location-cache kms-location-cache-key (string/upper-case location-string)))
+            _ (rl-util/log-redis-read-complete "lookup-by-location-string" kms-location-cache-key tm)]
+        (if keywords
+          keywords
+          (let [_ (load-cache-if-necessary context location-cache kms-location-cache-key)
+                [tm kwords] (util/time-execution (hash-cache/get-value location-cache kms-location-cache-key (string/upper-case location-string)))]
+            (rl-util/log-redis-read-complete "lookup-by-location-string" kms-location-cache-key tm)
+            kwords))))
+    (catch Exception e
+      (if (clojure.string/includes? (ex-message e) "Carmine connection error")
+        (do
+          (error "lookup-by-location-string found redis carmine exception. Will return nil result." e)
+          nil)
```

Comment:
Remove the (do and the nil)
(error returns nil.

Please do this in the kms_lookup namespace as well.